### PR TITLE
Fix esb-tooling-116

### PR DESCRIPTION
**Pull request by abeykoon**
```
<parameter name="max.store.connection.attempts">-1</parameter>
<parameter name="store.connection.retry.interval">1000</parameter>
<parameter name="message.processor.failMessagesStore">BackupMessageStore</parameter>

```

[1]. https://github.com/wso2/wso2-synapse/pull/1009

## Automation tests
 N/A
## Security checks
 N/A

## Samples
N/A

## Migrations (if applicable)
N/A

## Test environment
N/A
 
## Learning
N/A

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/model/MessageProcessorModel.java [hunk 1]
```diff
@@ -335,6 +364,12 @@ public Object getModelPropertyValue(String key) {
 				modelPropertyValue = getCronExpression();
 			} else if (key.equals("FS_processor.delivery_attempts")) {
 				modelPropertyValue = getDeliveryAttempts();
+			} else if(key.equals("FS_processor.max_store_connection_attempts")) {
```
Comment (malakaganga) [inline plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/model/MessageProcessorModel.java]: Formatting issue shall we fix this?

File: plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/validator/MessageProcessorFieldController.java [hunk 2]
```diff
@@ -84,6 +84,27 @@ public void validate(String modelProperty, Object value,
 							"Forwarding interval is not valid");
 				}
 			}
+		} else if(modelProperty.equals("FS_processor.max_store_connection_attempts")) {
```
Comment (malakaganga) [inline plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/validator/MessageProcessorFieldController.java]: Formatting issue shall we fix this?

File: plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/wizard/MessageProcessorCreationWizard.java [hunk 3]
```diff
@@ -229,6 +229,10 @@ private String getTemplateContent() {
 				messageProcessorPrameeters.put("max.delivery.attempts",
 						messageProcessorModel.getDeliveryAttempts());
 			}
+			
+			messageProcessorPrameeters.put("max.store.connection.attempts", Integer.toString(messageProcessorModel.getMaxStoreConnectionAttempts()));
```
Comment (malakaganga) [inline plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/wizard/MessageProcessorCreationWizard.java]: Formatting issue(line length) shall we fix this?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb/src/org/wso2/developerstudio/eclipse/gmf/esb/impl/MessageProcessorImpl.java [hunk 4]
```diff
@@ -1201,6 +1264,69 @@ public void setTaskCount(String newTaskCount) {
...
+	/**
+	 * <!-- begin-user-doc -->
+	 * <!-- end-user-doc -->
+	 * @generated
+	 */
+	public void setStoreConnectionInterval(int newStoreConnectionInterval) {
+		int oldStoreConnectionInterval = storeConnectionInterval;
+		storeConnectionInterval = newStoreConnectionInterval;
+		if (eNotificationRequired())
+			eNotify(new ENotificationImpl(this, Notification.SET, EsbPackage.MESSAGE_PROCESSOR__STORE_CONNECTION_INTERVAL, oldStoreConnectionInterval, storeConnectionInterval));
```
Comment (malakaganga) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb/src/org/wso2/developerstudio/eclipse/gmf/esb/impl/MessageProcessorImpl.java]: Formatting issue(line length) shall we fix this?

File: plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/model/MessageProcessorModel.java [hunk 5]
```diff
@@ -405,6 +440,12 @@ public boolean setModelPropertyValue(String key, Object data) throws ObserverFai
 			setCronExpression(data.toString());
 		} else if (key.equals("FS_processor.delivery_attempts")) {
 			setDeliveryAttempts(data.toString());
+		} else if(key.equals("FS_processor.max_store_connection_attempts")) {
+			setMaxStoreConnectionAttempts(Integer.parseInt(data.toString()));
+		} else if(key.equals("FS_processor.store_connection_retry_interval")) {
+			setStoreConnectionInterval(Integer.parseInt(data.toString()));
+		} else if(key.equals("FS_processor.fail_messages_store")) {
```
Comment (vinok88) [inline plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/model/MessageProcessorModel.java]: Fix the formatting issues.

File: plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/model/MessageProcessorModel.java (hunk 1 above)
Comment (abeykoon) [reply plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/model/MessageProcessorModel.java]: Fixed.

File: plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/validator/MessageProcessorFieldController.java (hunk 2 above)
Comment (abeykoon) [reply plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/validator/MessageProcessorFieldController.java]: Fixed.

File: plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/wizard/MessageProcessorCreationWizard.java (hunk 3 above)
Comment (abeykoon) [reply plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/wizard/MessageProcessorCreationWizard.java]: Need to keep it as it is as to fix this need to reformat the whole code.

File: plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/model/MessageProcessorModel.java (hunk 5 above)
Comment (abeykoon) [reply plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/model/MessageProcessorModel.java]: Fixed.

**Comment by prabushi** [comment]
Please add the newly added properties to esb.xsd file.

File: plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/provider/EmptyItemSelectedMessageStoreList.java [hunk 6]
```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Nirothipan) [inline plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/provider/EmptyItemSelectedMessageStoreList.java]: Year need to 2018 ?

**Comment by vinok88** [comment]
Please add a proper commit message.

File: plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/provider/EmptyItemSelectedMessageStoreList.java (hunk 6 above)
Comment (abeykoon) [reply plugins/org.wso2.developerstudio.eclipse.artifact.messageprocessor/src/org/wso2/developerstudio/eclipse/artifact/messageprocessor/provider/EmptyItemSelectedMessageStoreList.java]: Yes. I'll update the year.

**Pull request merged**